File: web/src/app/modules/game-data/model/question.ts
import {Action} from './action';

export class Question {
  id: number;
  amount: number;
  silver: boolean;
  question: string;
  answer: string;
  state: Action;
  pictureUri: string = '';
  soundUri: string = '';

  constructor(id: number,
              amount: number,
              isSilver: boolean,
              question: string,
              answer: string,
              pictureUri?: string,
              sound?: string) {
    this.id = id;
    this.amount = amount;
    this.silver = isSilver;
    this.question = question;
    this.answer = answer;
    this.state = Action.CLOSE;
    this.pictureUri = pictureUri;
    this.soundUri = sound;
  }
}

File: core/src/main/java/eu/kotkas/villak/core/model/Question.java
package eu.kotkas.villak.core.model;

import eu.kotkas.villak.core.model.enums.QuestionState;
import eu.kotkas.villak.core.util.IdUtil;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

/**
 * @author Kristen Kotkas
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Question implements Serializable {

  @Builder.Default
  private int id = IdUtil.getId();
  private int amount;
  private boolean silver;
  private String question;
  private String answer;
  @Builder.Default
  private QuestionState state = QuestionState.CLOSE;
  private String pictureUri;
  private String soundUri;

}
